File: microservice-product/src/main/java/com/products/application/message/PriceUpdatedMessage.java
package com.products.application.message;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Optional;

public record PriceUpdatedMessage(
        String sku,
        Optional<PriceMessage> basePrice,
        Optional<PriceMessage> currentPrice
) implements Serializable {

    public static record PriceMessage(
            String label,
            BigDecimal value
    ){}
}

File: microservice-pricing/src/main/kotlin/com/pricing/application/dto/PriceUpdatedMessage.kt
package com.pricing.application.dto

import java.math.BigDecimal

data class PriceUpdatedMessage(
    val sku: String,
    val basePrice: PriceMessage?,
    val currentPrice: PriceMessage?,
){
    data class PriceMessage (val label: String, val value: BigDecimal)
}
